app/services/fees/fixed_charge_service.rb

```diff
@@ -46,6 +54,20 @@ def already_billed?
     end
 
     def init_fee
+      # NOTE: Build fee for case when there is adjusted fee and units or amount has been adjusted.
+      # Base fee creation flow handles case when only name has been adjusted
+      if !current_usage && invoice&.draft? && adjusted_fee && !adjusted_fee.adjusted_display_name?
+        adjustment_result = Fees::InitFromAdjustedFixedChargeFeeService.call(
+          adjusted_fee:,
+          boundaries:,
+          properties: fixed_charge.properties
+        )
+        return result.fail_with_error!(adjustment_result.error) unless adjustment_result.success?
+
+        result.fee = adjustment_result.fee
+        return
+      end
```

Comment:
I'd suggest moving it into a separate method , so can have this method at least a little shorter 😔 
```
if !current_usage && invoice&.draft? && adjusted_fee && !adjusted_fee.adjusted_display_name?
  return handle_adjusted_fee
end
...
def handle_adjusted_fee
  adjustment_result = Fees::InitFromAdjustedFixedChargeFeeService.call(
          adjusted_fee:,
          boundaries:,
          properties: fixed_charge.properties
        )
        return result.fail_with_error!(adjustment_result.error) unless adjustment_result.success?

        result.fee = adjustment_result.fee
        return result
end
```